homeassistant/components/switch/wemo.py

```diff
@@ -54,29 +58,44 @@ def __init__(self, device):
         self.maker_params = None
         self.coffeemaker_mode = None
         self._state = None
+        self._available = True
+        self._update_lock = None
         # look up model name once as it incurs network traffic
         self._model_name = self.wemo.model_name
 
         wemo = get_component('wemo')
         wemo.SUBSCRIPTION_REGISTRY.register(self.wemo)
-        wemo.SUBSCRIPTION_REGISTRY.on(self.wemo, None, self._update_callback)
+        wemo.SUBSCRIPTION_REGISTRY.on(
+            self.wemo, None, self._subscription_callback)
 
-    def _update_callback(self, _device, _type, _params):
+    def _subscription_callback(self, _device, _type, _params):
         """Update the state by the Wemo device."""
-        _LOGGER.info("Subscription update for  %s", _device)
+        _LOGGER.info("Subscription update for %s", self.name)
         updated = self.wemo.subscription_update(_type, _params)
-        self._update(force_update=(not updated))
 
         if not hasattr(self, 'hass'):
```

Comment:
This can be removed after the move of callback registration to `async_added_to_hass`.